ui/apos/components/AposExportPiecesModal.vue

```diff
@@ -0,0 +1,149 @@
+<template>
+  <AposModal
+    :modal="modal"
+    class="apos-export-pieces"
+    v-on="{ esc: cancel }"
+    @no-modal="$emit('safe-close')"
+    @inactive="modal.active = false"
+    @show-modal="modal.showModal = true"
+    @ready="ready"
+  >
+    <template #main>
+      <AposModalBody>
+        <template #bodyMain>
+          <h2 class="apos-export-pieces__heading">
+            {{ $t('aposImportExport:export') }} {{ moduleName }}
+          </h2>
+          <p
+            class="apos-export-pieces__description"
+          >
+            {{ $t('aposImportExport:exportModalDescription', { count, type: moduleName }) }}
+          </p>
+          <div class="apos-export-pieces__btns">
+            <AposButton
+              class="apos-export-pieces__btn"
+              label="apostrophe:cancel"
+              @click="cancel"
+            />
+            <AposButton
+              ref="exportPieces"
+              class="apos-export-pieces__btn"
+              label="aposImportExport:export"
+              type="primary"
+              @click="exportPieces"
+            />
+          </div>
+        </template>
+      </AposModalBody>
+    </template>
+  </AposModal>
+</template>
+
+<script>
+export default {
+  props: {
+    moduleName: {
+      type: String,
+      default: ''
+    },
+    count: {
+      type: Number,
+      default: 1
+    }
+  },
+  emits: [ 'safe-close', 'modal-result' ],
+  data() {
+    return {
+      modal: {
+        active: false,
+        showModal: false,
+        disableHeader: true
+      },
+      formValues: null
+    };
+  },
+
+  async mounted() {
+    this.modal.active = true;
+  },
+  methods: {
+    ready() {
```

Comment:
This might be automatic soon when we switch to the `dialog` element, but the code is fine.